ui/static/ui/js/utils.jsx

```diff
@@ -116,6 +116,109 @@ define("utils", ["jquery", "lodash", "react", "react_infinite", "select2"],
   });
 
   /**
+  * Callback to send confirmation status.
+  *
+  * @callback confirmationResponse - Call back that sends confirmation status
+  * @param {bool} status - Confirmation status
+  */
+
+  /**
+   * Generic confirmation dialog.
+   * <ConfirmationDialog
+   *       id={confirmationDialogId}
+   *       actionButtonName={confirmationDialogActionButtonName}
+   *       title={confirmationDialogTitle}
+   *       message={confirmationDialogMessage}
+   *       confirmationSuccess={confirmationSuccess} />,
+   *
+   * @param {String} id - Bootstrap dialog id
+   * @param {String} actionButtonName - Name of action button can be any of delete, remove, ok etc
+   * @param {String} actionButtonClass - Class(es) for action button
+   * @param {String} title - Title/heading of confirmation dialog
+   * @param {String} message - Content of dialog
+   * @param {confirmationResponse} confirmationSuccess - Call back that sends confirmation status
+  */
+  var ConfirmationDialog =  React.createClass({
+    getInitialState: function() {
+      return {
+        id: 'confirm-box',
+        className: 'modal fade',
+        actionButtonName: 'Confirm',
+        actionButtonClass: 'btn btn-ok'
+      };
+    },
+    componentWillMount: function() {
+      if (this.props.actionButtonName) {
+        this.setState({actionButtonName: this.props.actionButtonName});
+      }
+
+      if (this.props.actionButtonClass) {
+        this.setState({actionButtonClass: this.props.actionButtonClass});
+      }
+
+      if (this.props.id) {
+        this.setState({id: this.props.id});
+      }
+
+      if (this.props.className) {
+        this.setState({className: this.props.className});
+      }
+    },
+    render: function() {
+      if (!this.props || !this.props.message) {
+        return null;
```

Comment:
This will cause problems with the `.modal('show')` in `componentDidMount` since there will be nothing to show. It would be cleaner to leave out this if statement and just display nothing if nothing is passed. If the caller intended the modal not to show up at all they should not instantiate this React element in the first place.